Microsoft.TeamServices.Samples.Client/WorkItemTracking/QueriesSample.cs

```diff
@@ -12,24 +13,401 @@ public class QueriesSample : ClientSample
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            List<QueryHierarchyItem> queries = workItemTrackingClient.GetQueriesAsync(projectId, QueryExpand.None, 1, true).Result;
+
+            if (queries == null)
+            {
+                Console.WriteLine("No queries found");
+            }
+            else
+            {
+                Console.WriteLine("Queries:");
+
+                foreach (var query in queries)
+                {
+                    Console.WriteLine("  {0} - {1}", query.Name, query.Path);
+                }
+            }
+
+            return queries;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem GetQueryOrFolderById()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryId = "2614c4de-be48-4735-9fdc-9656f55c495f";
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            QueryHierarchyItem query = workItemTrackingClient.GetQueryAsync(projectId, queryId).Result;
+
+            if (query == null)
+            {
+                Console.WriteLine("Query not found");
+            }
+            else
+            {
+                Console.WriteLine("Id:         {0}", query.Id);
+                Console.WriteLine("Name:       {0}", query.Name);
+                Console.WriteLine("Path:       {0}", query.Path);         
+            }
+
+            return query;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem GetQueryOrFolderByName()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryName = "Shared Queries/Current Iteration/Active Bugs";
 
             VssConnection connection = Context.Connection;
             WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
 
-            QueryHierarchyItem query = workItemTrackingClient.GetQueryAsync(project, queryName).Result;
+            QueryHierarchyItem query = workItemTrackingClient.GetQueryAsync(projectId, queryName).Result;
+
+            if (query == null)
+            {
+                Console.WriteLine("Query not found");
+            }
+            else
+            {
+                Console.WriteLine("Id:         {0}", query.Id);
+                Console.WriteLine("Name:       {0}", query.Name);
+                Console.WriteLine("Path:       {0}", query.Path);              
+            }
+
+            return query;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem GetDeletedQueryOrFolderById()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryId = "2614c4de-be48-4735-9fdc-9656f55c495f";
```

Comment:
Same comment as above --- tuck away a real query ID so this sample runs when any user runs it (unless this is a well-known ID that is guaranteed to be in any project).